alica_test_utility/include/alica/test/TestContext.h

```diff
@@ -0,0 +1,150 @@
+#pragma once
+
+#include <engine/AlicaContext.h>
+
+#include "BehaviourTrigger.h"
+
+#include <engine/AlicaEngine.h>
+#include <engine/model/Behaviour.h>
+#include <engine/model/EntryPoint.h>
+#include <engine/model/PlanType.h>
+#include <engine/model/Task.h>
+#include <engine/model/Variable.h>
+
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <string>
+
+namespace alica::test
+{
+class TestBehaviourCreator;
+class BehaviourTrigger;
+class TestContext : public alica::AlicaContext
+{
+public:
+    TestContext(const std::string& roleSetName, const std::string& masterPlanName, bool stepEngine, const essentials::IdentifierConstPtr agentID = nullptr);
+    /**
+     * This allows to get hold of the AlicaEngine, which is
+     * an inaccessible member variable of the AlicaContext, otherwise.
+     * @return Pointer to the AlicaEngine.
+     */
+    alica::AlicaEngine* getEngine();
+
+    /**
+     * Allows to retrieve the name of an AlicaElement.
+     * @param elementID The ID of the AlicaElement, whose name is requested.
+     * @return The name of the element.
+     */
+    template <typename AlicaElementType>
+    std::string getName(uint64_t elementID);
+
+    /**
+     * This method requests the engine to step until the given state is reached, or the
+     * time is over.
+     * @param state The state that the AlicaContext should reach before the timeout.
+     * @param msTimeout The timeout in milliseconds
+     * @return True, if the given state was reached before the timeout. False, otherwise.
+     */
+    template <typename Rep, typename Period>
+    bool stepUntilStateReached(int64_t state, std::chrono::duration<Rep, Period> timeout);
```

Comment:
consider the case: A single step of the engine can transition through multiple states.

Reply:
Well, if the engine jumps through multiple states, I expect this function to return false, because the agent did not stay in the requested state. Should I rename it to stepUntilStayInState?

Reply:
Do we have a requirement for this? if yes, maybe we can overwrite MaxRuleApplications = 1 while in test mode and then this API should probably work.

Reply:
This method semantically originates from sootballs testcases. So either there is a requirement for this, or they were not aware of ALICA being able to skip several state at once and at the same time did not encounter any problem with that.